cara/apps/calculator/__init__.py

```diff
@@ -129,6 +129,37 @@ async def post(self):
         self.finish(report)
 
 
+class ConcentrationModelJsonResponse(BaseRequestHandler):
+    async def post(self):
+        """
+        Expects algorithm input in HTTP POST request body in JSON format.
+        Returns report data (algorithm output) in HTTP POST response body in JSON format.
+        """
+        requested_model_config = json.loads(self.request.body)
+        if self.settings.get("debug", False):
+            from pprint import pprint
+            pprint(requested_model_config)
+
+        try:
+            form = model_generator.FormData.from_dict(requested_model_config)
```

Comment:
Is your objective really to allow the same **form** inputs to generate a report as well as JSON? (as implemented here)

I wonder if what you really want is a raw Model type (not ``FormData``, but an actual serialised ``cara.monte_carlo.models.ExposureModel`` instance)?

Reply:
The intention, as I understand the discussion with ARVE, is indeed to allow for exactly the same inputs that are specified through the HTML form. 

From a conceptual point of view, indeed, the detour over `FormData` might not be needed. In the end, the JSON endpoint "builds" a `cara.monte_carlo.models.ExposureModel` from the `FormData`. However, converting the HTTP request body into a `cara.monte_carlo.models.ExposureModel` instance would duplicated existing logic on `FormData`. This logic could be extracted into an own "entity" and re-used. However, in the end the data in the request body represents form data. I guess it could be OK to convert it into a `FormData` instance first. Or maybe rename `FormData` into `ModelData`?

Reply:
The intention is to generate the form payload from within the ARVE application automatically. The application runs on an embedded device that is placed in a room (e.g. hotel or conference room). The device contains several sensors, which measure air quality etc. The idea is then to generate the form payload from 

- the "configuration" of the room the device is place in
- sensor data of the device

Reply:
:+1: - thanks for the context.